patches/tModLoader/Terraria/GameContent/UI/States/UIAchievementsMenu.TML.cs

```diff
@@ -0,0 +1,122 @@
+		areYouSureText.HAlign = 0.5f;
+		areYouSureText.SetPadding(13f);
+		areYouSureText.Top.Set(-33, 0f);
+
+		areYouSureText.BackgroundColor = new Color(73, 94, 171);
+		achievementResetAreYouSure.Append(areYouSureText);
+
+		string text = FontAssets.ItemStack.Value.CreateWrappedText(Language.GetOrRegister("tModLoader.AchievementResetConfirmTooltip", () => "Resetting your achievements means you'll lose them permanently, are you sure you want to continue?").Value, 310, Language.ActiveCulture.CultureInfo);
+		UITextPanel<string> areYouSureDesciption = new UITextPanel<string>(text, 1f, large: false);
+		areYouSureDesciption.HAlign = 0.5f;
+		areYouSureDesciption.Top.Set(20, 0f);
+		areYouSureDesciption.SetPadding(13f);
+		areYouSureDesciption.Width.Set(-10, 1);
+		areYouSureDesciption.Height.Set(-50, 0.9f);
+		achievementResetAreYouSure.Append(areYouSureDesciption);
+
+		// Confirm Button
+		UITextPanel<LocalizedText> yesButton = new UITextPanel<LocalizedText>(Language.GetOrRegister("tModLoader.AchievementsReset", () => "Reset"), 0.7f, large: true);
+		yesButton.Width.Set(0, 0.5f);
+		yesButton.Height.Set(40f, 0f);
+		yesButton.VAlign = 1;
+		yesButton.HAlign = 1;
+		yesButton.OnMouseOver += FadedMouseOver;
+		yesButton.OnMouseOut += FadedMouseOut;
+		yesButton.OnLeftClick += ResetAchievements;
+		achievementResetAreYouSure.Append(yesButton);
+
+		// Cancel Button
+		UITextPanel<LocalizedText> noButton = new UITextPanel<LocalizedText>(Language.GetText("UI.Cancel"), 0.7f, large: true);
+		noButton.Width.Set(-10, 0.5f);
+		noButton.Height.Set(40f, 0f);
+		noButton.VAlign = 1;
+		noButton.OnMouseOver += FadedMouseOver;
+		noButton.OnMouseOut += FadedMouseOut;
+		noButton.OnLeftClick += CloseAchievementConfirm;
+		achievementResetAreYouSure.Append(noButton);
+	}
+
+	private void CloseAchievementConfirm(UIMouseEvent evt, UIElement listeningElement)
+	{
+		RemoveChild(blockingPanel);
+		RemoveChild(achievementResetAreYouSure);
+		blockingPanel = null;
+		achievementResetAreYouSure = null;
+	}
+
+	private void BlockInput(UIMouseEvent evt, UIElement listeningElement)
+	{
+	}
+
+	private void FilterSearch(object sender, EventArgs e)
+	{
+		_achievementsList.Clear();
+		string searchText = uISearchBar.Text?.ToLowerInvariant() ?? string.Empty; // Get the search text, ensuring it's lowercase
+
+		foreach (UIAchievementListItem achievementElement in _achievementElements) {
+			string friendlyName = achievementElement.GetAchievement().FriendlyName.Value.ToLowerInvariant(); // Convert to lowercase for case-insensitive comparison
+			string modName = achievementElement.GetAchievement().ModAchievement != null ? achievementElement.GetAchievement().ModAchievement.Mod.DisplayName.ToLowerInvariant() : string.Empty; // Convert to lowercase for case-insensitive comparison
+
+			if (friendlyName.Contains(searchText) || modName.Contains(searchText)) {
```

Comment:
You can use `StringComparison.CurrentCultureIgnoreCase` (current culture is probably better than invariant, right @JavidPack) 

https://learn.microsoft.com/en-us/dotnet/api/system.string.contains?view=net-9.0#system-string-contains(system-char-system-stringcomparison)